File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
import heapq

def solution(n, works):
    answer = 0
    if sum(works) < n:
        return 0
    
    hq = []
    for w in works:
        heapq.heappush(hq, -w)
    for i in range(n):
        t = heapq.heappop(hq)
        t += 1
        heapq.heappush(hq, t)
    
    for num in hq:
        answer += (num*num)
    return answer

File: 프로그래머스/3/12927. 야근 지수/야근 지수.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

long long solution(int n, vector<int> works) {
    priority_queue<int> pq;
    
    for (int work : works) {
        pq.push(work);
    }
    
    for (int i = 0; i < n; i++) {
        if (pq.empty() || pq.top() == 0) {
            break;  
        }
        
        int maxWork = pq.top();
        pq.pop();
        pq.push(maxWork - 1);  
    }
    
    long long answer = 0;
    while (!pq.empty()) {
        long long work = pq.top();
        pq.pop();
        answer += work * work;
    }
    
    return answer;
}